### PR TITLE
add support for libvirt provider

### PR DESCRIPTION
**Pull request by jforman**
libvirt provider support to be added to coreos-metadata in
https://github.com/coreos/coreos-metadata/pull/81.

### CODE REVIEW

**Comment by jforman** [comment]
Ping? What else should I do to get this reviewed and potentially committed? Thanks!

File: config/templating/templating.go [hunk 1]
```diff
@@ -58,6 +58,10 @@ var platformTemplatingMap = map[string]map[string]string{
 		fieldV4Private: "COREOS_GCE_IP_LOCAL_0",
 		fieldV4Public:  "COREOS_GCE_IP_EXTERNAL_0",
 	},
+	platform.Libvirt: {
+		fieldHostname:  "LIBVIRT_HOSTNAME",
```
Comment (lucab) [inline config/templating/templating.go]: `COREOS_LIBVIRT_HOSTNAME`

File: config/templating/templating.go [hunk 2]
```diff
@@ -58,6 +58,10 @@ var platformTemplatingMap = map[string]map[string]string{
...
+		fieldV4Private: "LIBVIRT_PRIVATE_IPV4",
```
Comment (lucab) [inline config/templating/templating.go]: `COREOS_LIBVIRT_IPV4_LOCAL`

**Comment by lucab** [comment]
Naming aside, this introduces a new cloud platform and also depends on a new coreos-metadata feature, thus I'd like to block this until coreos/coreos-metadata#81 had undergone its full review/merge/release cycle.

**Comment by sdemos** [comment]
given that this commit relies on a new cloud provider that isn't going to be merged into `coreos-metadata` per coreos/coreos-metadata#81, I'm going to close this.

**Pull request closed without merging**